staging/src/k8s.io/apiserver/pkg/authentication/serviceaccount/util.go

```diff
@@ -36,6 +36,27 @@ func MakeUsername(namespace, name string) string {
 	return ServiceAccountUsernamePrefix + namespace + ServiceAccountUsernameSeparator + name
 }
 
+// MatchesUsername checks whether the provided username matches the namespace and name without
+// allocating. Use this when checking a service account namespace and name against a known string.
+func MatchesUsername(namespace, name string, username string) bool {
+	if !strings.HasPrefix(username, ServiceAccountUsernamePrefix) {
```

Comment:
can avoid calling HasPrefix() twice (once directly, once indirectly via TrimPrefix) by taking the substring yourself once you pass HasPrefix checks.

Reply:
It's not super expensive in this case because most of the prefix is in the cache.  The cost is the allocation rather than the scan.

Reply:
The less ugly way to check it would be to compare to the previous length, but that makes the code harder to read.